File: programmers/해시/전화번호 목록(Level2).py
'''
    Code written on Aug 15, 2020
    출처: 프로그래머스 코딩 테스트 연습, https://programmers.co.kr/learn/challenges
    해시: 전화번호 목록(Level2)
    link: https://programmers.co.kr/learn/courses/30/lessons/42577
'''
from collections import deque
def solution(phone_book):
    phone_book.sort()
    for i, j in zip(phone_book, phone_book[1:]):
        if j.startswith(i):
            return False
    return True

File: programmers/해시/전화번호 목록(Level2).java
/*
    Code written on Sep 17, 2020
    출처: 프로그래머스 코딩 테스트 연습, https://programmers.co.kr/learn/challenges
    해시: 전화번호 목록(Level2)
    link: https://programmers.co.kr/learn/courses/30/lessons/42577
*/
import java.util.Arrays;

class Solution {
    public boolean solution(String[] phone_book) {
        Arrays.sort(phone_book);
        for (int i=1; i < phone_book.length; i++) {
            if (phone_book[i].startsWith(phone_book[i-1]))
                return false;
        }
        return true;
    }
}
